datumaro/components/media.py

```diff
@@ -72,8 +95,8 @@ def __init__(self,
         self._ext = ext
 
         if not isinstance(data, np.ndarray):
-            assert path or callable(data), "Image can not be empty"
-            assert data is None or callable(data)
+            assert path or callable(data) or size, "Image can not be empty"
+            assert data is None or callable(data) or size
```

Comment:
What's the use case for `Image` objects with _only_ the size filled in? Isn't the fundamental property of an image that it can be resolved to a bitmap?

Reply:
> Isn't the fundamental property of an image that it can be resolved to a bitmap?

Actually, it is not true even without this PR, because `path` can point to an inexistent file, so it can only be used as metainfo. And if the file is missing, but there is image size info in the format, we often try to construct a generic path, which such item would have. I think, that it is not totally correct, and we could just provide size metainfo in such cases. But I'm not sure. Maybe, it's worth to leave unchanged, until we have more strong reasons.

Reply:
Moreover, guessing the path, we can use another image (check the VOC import).

Reply:
I was going to write a rebuttal, but I think I actually found a use case where having the image dimensions is better than not having anything: you can use image dimensions when saving annotations in a format that requires normalized box coordinates.

So I guess there is some use for it, although I'm not a fan of how it complicates the data model...